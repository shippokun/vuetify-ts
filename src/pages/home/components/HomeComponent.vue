<template>
  <v-app id="inspire">
    <v-navigation-drawer v-model="drawer" app>
      <v-list dense>
        <v-list-item link>
          <v-list-item-action>
            <v-icon>mdi-home</v-icon>
          </v-list-item-action>
          <v-list-item-content>
            <v-list-item-title>Home</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
        <v-list-item link>
          <v-list-item-action>
            <v-icon>mdi-email</v-icon>
          </v-list-item-action>
          <v-list-item-content>
            <v-list-item-title>Contact</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-navigation-drawer>

    <v-main>
      <v-container class="fill-height" fluid>
        <v-row align="center" justify="center">
          <v-col class="text-center">
            <v-tooltip left>
              <template v-slot:activator="{ on }">
                <v-btn :href="source" icon large target="_blank" v-on="on">
                  <v-icon large>mdi-code-tags</v-icon>
                </v-btn>
              </template>
              <span>Source</span>
            </v-tooltip>
          </v-col>
        </v-row>
      </v-container>
    </v-main>
  </v-app>
</template>

<script lang="ts">
// @ is an alias to /src
import { Component, Prop, Emit, Vue } from 'vue-property-decorator';

@Component
export default class Home extends Vue {
  @Prop()
  public source?: string;

  @Prop()
  public draw!: boolean;

  public get drawer(): boolean {
    return this.draw;
  }

  public set drawer(value: boolean) {
    this.emit(value);
  }

  @Emit()
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  public emit(value: boolean) {}

  public emitToggle(): void {
    this.emit(!this.draw);
  }
}
</script>
