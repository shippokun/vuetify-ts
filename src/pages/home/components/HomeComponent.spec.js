import { shallowMount } from '@vue/test-utils';
import HomeComponent from './HomeComponent.vue';

describe('HomeComponent.vue', () => {
  const wrapper = shallowMount(HomeComponent, {
    propsData: {
      source: '',
    },
  });
  it('render', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
