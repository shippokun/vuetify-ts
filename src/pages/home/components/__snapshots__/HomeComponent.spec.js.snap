// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent.vue render 1`] = `
<v-main-stub tag="main">
  <v-container-stub tag="div" fluid="true" class="fill-height">
    <v-row-stub tag="div" align="center" justify="center">
      <v-col-stub tag="div" class="text-center">
        <v-tooltip-stub opendelay="0" closedelay="0" contentclass="" fixed="true" left="true" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" tag="span"> <span>Source</span></v-tooltip-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</v-main-stub>
`;
